tests/providers/cncf/kubernetes/test_pod_generator.py

```diff
@@ -170,7 +170,7 @@ def test_gen_pod_extract_xcom(self, mock_rand_str, data_file):
         template_file = data_file("pods/generator_base_with_secrets.yaml").as_posix()
 
         pod_generator = PodGenerator(pod_template_file=template_file, extract_xcom=True)
-        with pytest.warns(AirflowProviderDeprecationWarning, match="deprecated and will be removed"):
+        with pytest.warns(AirflowProviderDeprecationWarning):
```

Comment:
We don't need to match the error. 

If we do, it fails because the last warning is raised from  `self.add_xcom_sidecar(result)` which is also deprecated.

https://github.com/apache/airflow/blob/f5c86edb7967102d26435018e367354735044f56/airflow/providers/cncf/kubernetes/pod_generator.py#L169

https://github.com/apache/airflow/blob/f5c86edb7967102d26435018e367354735044f56/airflow/providers/cncf/kubernetes/pod_generator.py#L173-L181